sem_covid/adapters/embedding_models.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# Date    : 20.07.2021 
+# Author  : Stratulat Ștefan
+# File    : embedding_models.py
+# Software: PyCharm
+from typing import List
+
+import numpy as np
+from gensim.models import KeyedVectors
+
+from sem_covid.adapters.abstract_model import (WordEmbeddingModelABC, SentenceEmbeddingModelABC,
+                                               TokenizerModelABC)
+import tensorflow_hub as hub
+
+import pandas as pd
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+from transformers import AutoTokenizer, AutoModel, BertModel, BertTokenizerFast
+
+
+class BasicTokenizerModel(TokenizerModelABC):
+
+    def tokenize(self, text: str) -> List[str]:
+        return text.split(' ')
+
+
+class SpacyTokenizerModel(TokenizerModelABC):
+
+    def __init__(self, spacy_nlp):
+        self._spacy_nlp = spacy_nlp
+
+    def tokenize(self, text: str) -> List[str]:
+        return list(map(str, self._spacy_nlp(text)))
+
+
+class Word2VecEmbeddingModel(WordEmbeddingModelABC):
+
+    def __init__(self, word2vec: KeyedVectors):
+        self._word2vec = word2vec
+        self._vector_dimension = word2vec.vector_size
+
+    def encode(self, tokens: List[str]) -> List:
+        return [self._word2vec[word].tolist() if word in self._word2vec else np.zeros(self._vector_dimension).tolist()
+                for word in tokens]
+
+
+class AverageSentenceEmbeddingModel(SentenceEmbeddingModelABC):
+
+    def __init__(self, word_embedding_model: WordEmbeddingModelABC, tokenizer: TokenizerModelABC):
+        self._word_embedding_model = word_embedding_model
+        self._tokenizer = tokenizer
+
+    def encode(self, sentences: List[str]) -> List:
```

Comment:
Let's also add nice and explicit documentation of what this class is intended for.

I believe here we want that the list of `tokens` is in fact a set of consecutive tokens of a sentence or a paragraph or a document. 

Possibly as are in the text or possibly tokens after the stop-words and punctuation were removed. Having this list instead of the original text is an advantage because we have externalised tokenisation and pre-processing.